src/compile/data.ts

```diff
@@ -23,79 +23,228 @@ const DEFAULT_NULL_FILTERS = {
 };
 
 /**
- * Create Vega's data array from a given model.
+ * Composable component instance of a model's data.
+ */
+export interface DataComponent {
```

Comment:
Signature for each member of DataComponent might get revised more as we continue implementing composition.